build/utils.py

```diff
@@ -0,0 +1,74 @@
+# Copyright The FMS HF Tuning Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Standard
+import logging
+
+# Third Party
+import transformers
+
+# Local
+from tuning.config import configs, peft_config
+
+
+def process_launch_training_args(job_config_dict):
+    """Return parsed config for tuning to pass to SFT Trainer
+    Args:
+        job_config_dict: dict
+    Return:
+        model_args: configs.ModelArguments
+        data_args: configs.DataArguments
+        training_args: configs.TrainingArguments
+        tune_config: peft_config.LoraConfig | peft_config.PromptTuningConfig
+        merge_model: bool
+    """
+    parser = transformers.HfArgumentParser(
+        dataclass_types=(
+            configs.ModelArguments,
+            configs.DataArguments,
+            configs.TrainingArguments,
+            peft_config.LoraConfig,
+            peft_config.PromptTuningConfig,
+        )
+    )
+    peft_method_parsed = "pt"
```

Comment:
hmm do we need to set this here to pt? I know it was done earlier in launch training too, but now I am confused why.. @anhuong ?
we are reading it below - peft_method_parsed = job_config_dict.get("peft_method")

and if not set, it should remain None for full fine tuning

Reply:
I agree that this doensn't need to be set here. I think we used to set the peft_method to be PT by default but I know a recent PR went in to update that. In addition, as you said since we pull the value from the JSON, this gets updated to `None` appropriately. I verified in my tests that I ran which printed out the params parsed and passed into sft_trainer, that `tune_config` gets set to None appropriately